sklearn/feature_extraction/_hashing_fast.pyx

```diff
@@ -68,7 +68,11 @@ def transform(raw_X, Py_ssize_t n_features, dtype,
             h = murmurhash3_bytes_s32(<bytes>f, seed)
 
             array.resize_smart(indices, len(indices) + 1)
-            indices[len(indices) - 1] = abs(h) % n_features
+            if h == - 2147483648:
+                # abs(-2**31) is undefined behavior
+                indices[len(indices) - 1] = (2147483647 - (n_features - 1)) % n_features
```

Comment:
Is this mapping arbitrary? As in, can we have mapping the indices as follows:

```py
indices[len(indices) - 1] = 2147483647 % n_features
```

Reply:
It's arbitrary. I chose it to be equal to ``abs(h) % n_features`` even when abs(h) is undefined.

Reply:
Ah I see, we can include a comment to state this:

```python
# abs(-2**31) is undefined behavior because h is a `np.int32`
# The following is defined such that it is equal to: abs(-2**31) % n_features
```